packages/backend-deployer/src/cdk_deployer.ts

```diff
@@ -199,14 +199,23 @@ export class CDKDeployer implements BackendDeployer {
       // However if the cdk process didn't run or produced no output, then we have nothing to go on with. So we throw
       // this error to aid in some debugging.
       if (aggregatedStderr.trim()) {
+        // If the string is more than 65KB, truncate and keep the last portion.
         // eslint-disable-next-line amplify-backend-rules/prefer-amplify-errors
-        throw new Error(aggregatedStderr);
+        throw new Error(this.truncateString(aggregatedStderr, 65000));
```

Comment:
Would it make sense to do something like:

```
new Error(`${this.head(aggregatedStderr, 55000)}... truncated ... ${this.tail(aggregatedStderr)}`);
```
so that there's a chance to see fragment of root cause if it's at the end?

If we don't then w should at least indicate that message is truncated.
```
new Error(`${this.truncateString(aggregatedStderr, 65000)}... truncated`);
```

Reply:
Updated